### PR TITLE
feat: Add listPopFront API

### PR DESCRIPTION
**Pull request by rishtigupta**
## PR Description:
- Add listPopFront API
- Add integ tests

## Build:
`./gradlew :momento-sdk:spotlessApply && ./gradlew clean build`

## Issue
#170

### CODE REVIEW

File: momento-sdk/src/intTest/java/momento/sdk/ListTest.java [hunk 1]
```diff
@@ -569,4 +570,47 @@ public void shouldFailListPopBackWhenNullListName() {
...
+        .succeedsWithin(FIVE_SECONDS)
+        .isInstanceOf(CacheListFetchResponse.Miss.class);
+
+    assertThat(
+            target.listConcatenateBackString(
+                cacheName, listName, values, CollectionTtl.fromCacheTtl(), 0))
+        .succeedsWithin(FIVE_SECONDS)
+        .isInstanceOf(CacheListConcatenateBackResponse.Success.class);
+
+    // Pop the value as string from back of the list
```
Comment (nand4011) [inline momento-sdk/src/intTest/java/momento/sdk/ListTest.java:588]: Nit: these comments should say front

File: momento-sdk/src/main/java/momento/sdk/messages/CacheListPopFrontResponse.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+import com.google.protobuf.ByteString;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import momento.sdk.exceptions.SdkException;
+import momento.sdk.internal.StringHelpers;
+
+/** Response for a list pop front operation */
+public interface CacheListPopFrontResponse {
+
+  /** A successful list pop front operation that found elements. */
```
Comment (nand4011) [inline momento-sdk/src/main/java/momento/sdk/messages/CacheListPopFrontResponse.java:12]: Need to remove the plural.

**Review by nand4011 — COMMENTED** [review]
Looks good aside from a couple doc issues.

File: momento-sdk/src/main/java/momento/sdk/CacheClient.java [hunk 3]
```diff
@@ -616,6 +617,18 @@ public CompletableFuture<CacheListPopBackResponse> listPopBack(
     return scsDataClient.listPopBack(cacheName, listName);
   }
 
+  /**
+   * Fetches the value from the front of the given list.
```
[the remark this answers is not in the document]
Comment (rishtigupta) [reply momento-sdk/src/main/java/momento/sdk/CacheClient.java:621]: Yes it does, good catch. Thanks!

File: momento-sdk/src/main/java/momento/sdk/messages/CacheListPopFrontResponse.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+import com.google.protobuf.ByteString;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import momento.sdk.exceptions.SdkException;
+import momento.sdk.internal.StringHelpers;
+
+/** Response for a list pop front operation */
+public interface CacheListPopFrontResponse {
+
+  /** A successful list pop front operation that found element. */
```
Comment (nand4011) [inline momento-sdk/src/main/java/momento/sdk/messages/CacheListPopFrontResponse.java:12]: You need an 'an' in both of these comments, but that can be updated in the next pr.

**Review by nand4011 — APPROVED** [review]

**Pull request merged**